components/tooltip/script.js

```diff
@@ -0,0 +1,97 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Coherent Labs AD. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import components from 'coherent-gameface-components';
+import template from './template.html';
+const TOOLTIP_MARGIN = 10;
+
+class Tooltip extends HTMLElement {
+    constructor() {
+        super();
+        this.template = template;
+
+        this.state = {
+            visible: false
+        };
+    }
+
+    connectedCallback() {
+        this.position = this.getAttribute('position') || 'top';
+        this.showOn = this.getAttribute('on') || 'click';
+        this.hideOn = this.getAttribute('off') || 'click';
+        this.elementSelector = this.getAttribute('for');
+        // TODO: check if this.elementSelector is valid and attached to DOM
+        this.triggerElement = document.querySelector(this.elementSelector);
+
+        components.loadResource(this)
+            .then((result) => {
+                this.template = result.template;
+                components.renderOnce(this);
+                this.attachEventListeners();
+            })
+            .catch(err => console.error(err));
+    }
+
+    attachEventListeners() {
+        // TODO: validate if showOn is a valid onEvent handler

```

Comment:
We can just remove this TODO because it will stop working with custom events if we make such a validation.

Reply:
Yes. I didn't think of this. Removed.